src/main/java/com/automation/school/ad_main/Collection/Company.java

```diff
@@ -0,0 +1,16 @@
+package com.automation.school.ad_main.Collection;
+
+import java.util.Map;
+
+public class Company {
+    private long id;
+    private String company;
+    private Map<String,Address> addressesCompany; //(i.e. Key= "Home", Value = Address)
+    private static int nextUniqueId = 0;
+
+    public int getCompanyId() {
+        nextUniqueId++;
+        id = nextUniqueId;
+        return (int) id;
```

Comment:
why didn't you keep the id as long in the return? Also the method return type

Reply:
Method  is rewritten  with this one 
public int getCompanyId() {
        return id++;
    }

Reply:
with such logic, as we have here - each time you call getCompanyId method - your id will change.
i don't think you're expecting this this to be happened
plus this class doesn't have neither all-args constructor nor setters, how would you set company name and addresses for example?)

Reply:
New class created for Objects count.

public class IdGenerator {
     public IdGenerator() {
    }
    public long set(long current) {
        return current++;
    }

Classes  was updated to use IdGenerator  class